torch/csrc/jit/python/pybind_utils.h

```diff
@@ -476,6 +476,17 @@ inline void guardAgainstNamedTensor(const T& var) {
       "workaround please drop names via `tensor = tensor.rename(None)`.");
 }
 
+// NB: When passing py::object as the first argument, the call site needs to
+// keep the py::objec alive until this function returns.
+//
+// Note [jit::toIValue barrow py::object refcnt]
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~
+// This function takes a py::handle as the first argument. So if the call site
+// passes a py:object as the first argument, this will be copy by value and the
+// py::handle is only live as long as py::object is, and the py::handle is not
+// reference counted. Hence, the call site needs to keep py::object alive until
+// a new copy is created by jit::toIValue().
```

Comment:
Could this just say "keep py::object alive until toIValue()" returns? The caller should not have to be responsible for knowing the internal details, right?

Reply:
Yes, let me modify that.